arch/x86/core/x86_mmu.c

```diff
@@ -17,9 +17,13 @@
 /* Mark text and rodata as read-only.
  * Userspace may read all text and rodata.
  */
-MMU_BOOT_REGION((u32_t)&_image_rom_start, (u32_t)&_image_rom_size,
+MMU_BOOT_REGION((u32_t)&_image_text_start, (u32_t)&_image_text_size,
 		MMU_ENTRY_READ | MMU_ENTRY_USER);
 
+MMU_BOOT_REGION((u32_t)&_image_rodata_start, (u32_t)&_image_rodata_size,
+		MMU_ENTRY_USER |
+		MMU_ENTRY_EXECUTE_DISABLE);;
```

Comment:
shouldn't keep the same semantic and do:
``MMU_ENTRY_READ | MMU_ENTRY_USER | MMU_ENTRY_EXECUTE_DISABLE```

Reply:
fixed. MMU_ENTRY_READ is a zero value which is why it worked without.